tests/test_butler.py

```diff
@@ -1193,5 +1202,117 @@ def tearDown(self):
             unsetAwsEnvCredentials()
 
 
+@unittest.skipIf(WsgiDAVApp is None, "Warning: wsgi/cheroot not found!")
+class WebdavDatastoreButlerTestCase(FileLikeDatastoreButlerTests, unittest.TestCase):
+    """WebdavDatastore specialization of a butler; a Webdav storage Datastore +
+    a local in-memory SqlRegistry.
+    """
+    configFile = os.path.join(TESTDIR, "config/basic/butler-webdavstore.yaml")
+    fullConfigKey = None
+    validationCanFail = True
+
+    serverName = "localhost:8080"
+    """Name of the server that will be used in the tests. The name is read from
+    the config file used with the tests during set-up.
+    """
+
+    root = "butlerRoot/"
+    """Root repository directory expected to be used in case useTempRoot=False.
+    Otherwise the root is set to a 20 characters long randomly generated string
+    during set-up.
+    """
+
+    datastoreStr = [f"datastore={root}"]
+    """Contains all expected root locations in a format expected to be
+    returned by Butler stringification.
+    """
+
+    datastoreName = ["WebdavDatastore@https://{serverName}/{root}"]
+    """The expected format of the WebdavDatastore string."""
+
+    registryStr = ":memory:"
+    """Expected format of the Registry string."""
+
+    def genRoot(self):
+        """Returns a random string of len 20 to serve as a root
+        name for the temporary bucket repo.
+
+        This is equivalent to tempfile.mkdtemp as this is what self.root
+        becomes when useTempRoot is True.
+        """
+        rndstr = "".join(
+            random.choice(string.ascii_uppercase + string.digits) for _ in range(20)
+        )
+        return rndstr + "/"
+
+    @classmethod
+    def setUpClass(cls):
+        # Do the same as inherited class
+        cls.storageClassFactory = StorageClassFactory()
+        cls.storageClassFactory.addFromConfig(cls.configFile)
+
+        # Run a local webdav server on which tests will be run
+        t = Thread(target=cls._serveWebdav, daemon=True)
+        t.start()
+        # Wait for it to start
+        time.sleep(3)
+
+    def setUp(self):
+        os.environ["WEBDAV_AUTH_METHOD"] = "TOKEN"
+        os.environ["WEBDAV_BEARER_TOKEN"] = "XXXXXX"
+
+        config = Config(self.configFile)
+        uri = ButlerURI(config[".datastore.datastore.root"])
+        self.serverName = uri.netloc
+
+        if self.useTempRoot:
+            self.root = self.genRoot()
+        rooturi = f"http://{self.serverName}/{self.root}"
+        config.update({"datastore": {"datastore": {"root": rooturi}}})
+
+        self.datastoreStr = f"datastore={self.root}"
+        self.datastoreName = [f"WebdavDatastore@{rooturi}"]
+
+        Butler.makeRepo(rooturi, config=config, forceConfigRoot=False)
```

Comment:
`makeRepo` can take a `ButlerURI` (but make sure you use `forceDirectory`) if that makes things easier for you.